src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1577,6 +1577,55 @@ def validate_azcopy_credential(cmd, namespace):
                                             service=service, resource_types='sco', permissions='rl')
 
 
+def is_directory(props):
+    return 'hdi_isfolder' in props.metadata.keys() and props.metadata['hdi_isfolder'] == 'true'
+
+
+def validate_fs_directory_upload_destination_url(cmd, namespace):
+    destination = namespace.destination_path
+    container = namespace.destination_fs
+    kwargs = {'account_name': namespace.account_name,
+              'account_key': namespace.account_key,
+              'connection_string': namespace.connection_string,
+              'sas_token': namespace.sas_token}
+    client = blob_data_service_factory(cmd.cli_ctx, kwargs)
+    destination = destination[1:] if destination.startswith('/') else destination
+    from azure.common import AzureException
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+    try:
+        props = client.get_blob_properties(container, destination)
```

Comment:
the same as above. please try to get properties with file datalake related SDK.

Reply:
done